Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool.swift

```diff
@@ -12,8 +12,114 @@
 //
 //===----------------------------------------------------------------------===//
 
+import NIO
+
 enum HTTPConnectionPool {
-    struct Connection {
+    struct Connection: Equatable {
         typealias ID = Int
+
+        private enum Reference {
+//            case http1_1(HTTP1Connection)
+
+            #if DEBUG
```

Comment:
We sometimes want to run tests in release mode: we should probably just define this always and name it to indicate its use in testing.